### PR TITLE
add metal kernels for norm, alibi, f16->f16 copy

### PR DESCRIPTION
**Pull request by apage43**
Tested with replit model, should also enable MPT and possibly others

### CODE REVIEW

File: ggml-metal.metal [hunk 1]
```diff
@@ -256,6 +256,72 @@ kernel void kernel_get_rows_q4_1(
...
+    }
+    // broadcast
+    if (tpitg == 0) {
+        sum[0] /= ne00;
+    }
+    threadgroup_barrier(mem_flags::mem_threadgroup);
+    const float mean  = sum[0];
+
+    // recenter
+    device float * y = dst + tgpig*ne00;
```
[the remark this answers is not in the document]
Comment (apage43) [reply ggml-metal.metal:292]: I've tried this but the results seem noticeably worse - at this point I'm not sure if I have a bug I can't find or if this is just a numerical precision problem

[the remark this answers is not in the document]
Comment (apage43) [reply ggml-metal.metal:292]: decided to check out what pytorch does - on Cuda they use Welford's algorithm avoid accuracy issues doing it in a single pass but on Metal they just use Metal Performance Shaders Graph's `variance` and `mean` ops

**Review by ggerganov — APPROVED** [review]

**Comment by ggerganov** [comment]
Let's resolve the conflicts and merge

**Comment by apage43** [comment]
conflicts resolved

**Pull request merged**